app/src/main/java/com/example/caculator_kotlin/Calculator.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.caculator_kotlin
+
+import java.util.*
+import java.util.regex.Pattern
+import kotlin.collections.ArrayList
+
+class Calculator {
+    companion object {
+        private const val parse_token_regex = "([()])|(\\d+(\\.\\d+)?)|([+*/-])"
```

Comment:
conventions về đặt tên biến 
https://kotlinlang.org/docs/coding-conventions.html#property-names
```suggestion
        private const val PARSE_TOKEN_REGEX = "([()])|(\\d+(\\.\\d+)?)|([+*/-])"
```